packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -42,11 +43,21 @@ export const CommerceApiContext = React.createContext({} as ApiClients)
  * @returns Provider to wrap your app with
  */
 const CommerceApiProvider = (props: CommerceApiProviderProps): ReactElement => {
-    const {children, clientId, organizationId, shortCode, siteId, proxy, queryClientConfig} = props
+    const {
+        children,
+        clientId,
+        organizationId,
+        shortCode,
+        siteId,
+        proxy,
+        queryClientConfig,
+        correlationId = ''
+    } = props
 
     // DEBUG: copy access token from browser
     const headers = {
-        authorization: ''
+        authorization: '',
+        ...(correlationId && {'x-correlation-id': correlationId})
```

Comment:
Sorry, dumb question: by applying this in a global manner, the correlation id will be reused on multiple requests?

Reply:
I've reached out to the IPA team who will be doing the external id support. We don't necessarily have to make it unique for each request, but instead, we want to have a new id when we navigate to a new page.